packages/app-js/src/Editor.tsx

```diff
@@ -7,74 +7,63 @@ import { BareProps } from '@polkadot/ui-app/types';
 import React from 'react';
 import CodeFlask from 'codeflask';
 
-import DEFAULT_CODE from './snippets/newHead';
 import WRAPPING from './snippets/wrapping';
 
 type Props = BareProps & {
   children?: React.ReactNode,
-  onEdit: (code: string) => void
-};
-type State = {
-  code: string
+  code: string,
+  onEdit: (code: string) => void,
+  snippet: string
 };
 
-export default class Editor extends React.PureComponent<Props, State> {
+export default class Editor extends React.PureComponent<Props> {
   private id: string;
+  private editor: any;
 
   constructor (props: Props) {
     super(props);
-
     this.id = `flask-${Date.now()}`;
-    this.state = {
-      code: `${WRAPPING}${DEFAULT_CODE}`
-    };
   }
 
   componentDidMount () {
-    const { onEdit } = this.props;
-    const { code } = this.state;
-
-    const editor = new CodeFlask(`#${this.id}`, {
+    // const { code } = this.props;
+    this.editor = new CodeFlask(`#${this.id}`, {
       language: 'js',
       lineNumbers: true
     });
 
-    editor.updateCode(code);
+    const { editor, props: { code, onEdit } } = this;
+
+    editor.updateCode(`${WRAPPING}${code}`);
 
     editor.onUpdate((code: string) => {
-      this.onEdit(code);
+      this.props.onEdit(code);
     });
 
     editor.editorRoot.addEventListener('focusin', () => {
-      editor.onUpdate(this.onEdit);
-    });
-
-    editor.editorRoot.addEventListener('focusout', () => {
-      editor.onUpdate(() => {
-        // empty
-      });
+      this.editor.onUpdate(this.props.onEdit);
     });
 
     onEdit(code);
   }
 
-  render () {
-    const { children } = this.props;
+  componentWillReceiveProps (nextProps: any) {
```

Comment:
Legacy, avoid. https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

`getDerivedStateFromProps` is the current recommended version

Reply:
I've rewritten it to getDerivedStateFromProps